Build failure: lxsession

## Steps To Reproduce

Steps to reproduce the behavior:

1. build lxsession

## Build log

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/inswlrh4qx8qhq92gjsqffvc0pnxf0a6-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/hv6697vi0dk7w5cwzipzd82khgh3yydx-xmlcatalog_patch.patch
patching file configure.ac
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
+ '[' x '!=' x ]
+ aclocal
+ autoheader --force
+ AUTOMAKE=automake intltoolize -c --automake --force
+ automake --add-missing --copy --include-deps
configure.ac:16: installing './compile'
configure.ac:4: installing './install-sh'
configure.ac:4: installing './missing'
Makefile.am: installing './depcomp'
+ autoconf
configure.ac:19: warning: The macro 'AC_PROG_CC_STDC' is obsolete.
configure.ac:19: You should run autoupdate.
./lib/autoconf/c.m4:1669: AC_PROG_CC_STDC is expanded from...
configure.ac:19: the top level
configure.ac:167: warning: The macro 'GLIB_GNU_GETTEXT' is obsolete.
configure.ac:167: You should run autoupdate.
aclocal.m4:388: GLIB_GNU_GETTEXT is expanded from...
aclocal.m4:488: AM_GLIB_GNU_GETTEXT is expanded from...
configure.ac:167: the top level
configure.ac:167: warning: The macro 'AC_TRY_LINK' is obsolete.
configure.ac:167: You should run autoupdate.
./lib/autoconf/general.m4:2918: AC_TRY_LINK is expanded from...
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
aclocal.m4:73: GLIB_LC_MESSAGES is expanded from...
aclocal.m4:388: GLIB_GNU_GETTEXT is expanded from...
aclocal.m4:488: AM_GLIB_GNU_GETTEXT is expanded from...
configure.ac:167: the top level
configure.ac:167: warning: The macro 'AC_TRY_LINK' is obsolete.
configure.ac:167: You should run autoupdate.
./lib/autoconf/general.m4:2918: AC_TRY_LINK is expanded from...
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/headers.m4:89: _AC_CHECK_HEADER_COMPILE is expanded from...
./lib/autoconf/headers.m4:56: AC_CHECK_HEADER is expanded from...
aclocal.m4:172: GLIB_WITH_NLS is expanded from...
aclocal.m4:388: GLIB_GNU_GETTEXT is expanded from...
aclocal.m4:488: AM_GLIB_GNU_GETTEXT is expanded from...
configure.ac:167: the top level
configure.ac:167: warning: The macro 'AC_TRY_LINK' is obsolete.
configure.ac:167: You should run autoupdate.
./lib/autoconf/general.m4:2918: AC_TRY_LINK is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/headers.m4:89: _AC_CHECK_HEADER_COMPILE is expanded from...
./lib/autoconf/headers.m4:56: AC_CHECK_HEADER is expanded from...
aclocal.m4:172: GLIB_WITH_NLS is expanded from...
aclocal.m4:388: GLIB_GNU_GETTEXT is expanded from...
aclocal.m4:488: AM_GLIB_GNU_GETTEXT is expanded from...
configure.ac:167: the top level
configure.ac:167: warning: The macro 'AC_OUTPUT_COMMANDS' is obsolete.
configure.ac:167: You should run autoupdate.
./lib/autoconf/status.m4:1025: AC_OUTPUT_COMMANDS is expanded from...
aclocal.m4:172: GLIB_WITH_NLS is expanded from...
aclocal.m4:388: GLIB_GNU_GETTEXT is expanded from...
aclocal.m4:488: AM_GLIB_GNU_GETTEXT is expanded from...
configure.ac:167: the top level
+ rm -rf autom4te.cache
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
configure flags: --disable-dependency-tracking --prefix=/nix/store/7ck29h16qqwnjr5f8c4y5lhx7f7gj9b9-lxsession-0.5.5 --enable-man --disable-buildin-clipboard --disable-buildin-polkit --with-xml-catalog=/nix/store/rzrppq273z5fsbv5kci1n8n353ahx7n1-docbook-xml-4.1.2/xml/dtd/docbook/catalog.xml
configure: WARNING: unrecognized options: --with-xml-catalog
checking for a BSD-compatible install... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether NLS is requested... yes
checking for intltool >= 0.40.0... 0.51.0 found
checking for intltool-update... /nix/store/5kg3f63gkpwgdawnv9hvygcdis4arjvg-intltool-0.51.0/bin/intltool-update
checking for intltool-merge... /nix/store/5kg3f63gkpwgdawnv9hvygcdis4arjvg-intltool-0.51.0/bin/intltool-merge
checking for intltool-extract... /nix/store/5kg3f63gkpwgdawnv9hvygcdis4arjvg-intltool-0.51.0/bin/intltool-extract
checking for xgettext... /nix/store/9a46kza0vbb4vdsdmpzbdc7f31y7hs0i-gettext-0.22.5/bin/xgettext
checking for msgmerge... /nix/store/9a46kza0vbb4vdsdmpzbdc7f31y7hs0i-gettext-0.22.5/bin/msgmerge
checking for msgfmt... /nix/store/9a46kza0vbb4vdsdmpzbdc7f31y7hs0i-gettext-0.22.5/bin/msgfmt
checking for gmsgfmt... /nix/store/9a46kza0vbb4vdsdmpzbdc7f31y7hs0i-gettext-0.22.5/bin/msgfmt
checking for perl... /nix/store/1i003ijlh9i0mzp6alqby5hg3090pjdx-perl-5.40.0/bin/perl
checking for perl >= 5.8.1... 5.40.0
checking whether make supports nested variables... (cached) yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... none
checking for mkstemp... yes
checking pkg-config is at least version 0.9.0... yes
checking for x11... yes
checking for glib-2.0 >= 2.28.0... yes
checking for gio-unix-2.0... yes
checking for gtk+-2.0 >= 2.6.0... yes
checking for polkit-agent-1... yes
checking for valac... /nix/store/v2kdxvkp36r7p9wagxbfm3qzhsqkcfyv-vala-0.56.17/bin/valac
checking whether /nix/store/v2kdxvkp36r7p9wagxbfm3qzhsqkcfyv-vala-0.56.17/bin/valac supports at least API version 0.16.0... yes
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for locale.h... yes
checking for LC_MESSAGES... yes
checking for CFPreferencesCopyAppValue... no
checking for CFLocaleCopyCurrent... no
checking for libintl.h... yes
checking for ngettext in libc... yes
checking for dgettext in libc... yes
checking for bind_textdomain_codeset... yes
checking for msgfmt... (cached) /nix/store/9a46kza0vbb4vdsdmpzbdc7f31y7hs0i-gettext-0.22.5/bin/msgfmt
checking for dcgettext... yes
checking if msgfmt accepts -c... yes
checking for gmsgfmt... (cached) /nix/store/9a46kza0vbb4vdsdmpzbdc7f31y7hs0i-gettext-0.22.5/bin/msgfmt
checking for xgettext... (cached) /nix/store/9a46kza0vbb4vdsdmpzbdc7f31y7hs0i-gettext-0.22.5/bin/xgettext
checking for xsltproc... /nix/store/ckqaamnjzciyl1wgmh99cjlpdjmzvaiz-libxslt-1.1.42-bin/bin/xsltproc
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating data/Makefile
config.status: creating data/ui/Makefile
config.status: creating data/lxpolkit.desktop.in
config.status: creating man/Makefile
config.status: creating po/Makefile.in
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing default-1 commands
config.status: executing po/stamp-it commands
configure: WARNING: unrecognized options: --with-xml-catalog

LXSession.................................... : Version 0.5.5

Prefix....................................... : /nix/store/7ck29h16qqwnjr5f8c4y5lhx7f7gj9b9-lxsession-0.5.5
Sysconfdir................................... : ${prefix}/etc
GTK support.................................. : yes
GTK Cliboard build-in suport................. : no
GTK Polkit-agent build-in suport............. : no
Enable GTK3 support.......................... : no
Enable debug................................. : no
Man pages generation support................. : yes

The binary will be installed in /nix/store/7ck29h16qqwnjr5f8c4y5lhx7f7gj9b9-lxsession-0.5.5/bin

http://lxde.org/

Have fun!

Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash
make  all-recursive
make[1]: Entering directory '/build/source'
Making all in data
make[2]: Entering directory '/build/source/data'
Making all in ui
make[3]: Entering directory '/build/source/data/ui'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/build/source/data/ui'
make[3]: Entering directory '/build/source/data'
  ITMRG  lxpolkit.desktop
  ITMRG  lxsession-default-apps.desktop
  ITMRG  lxsession-edit.desktop
make[3]: Leaving directory '/build/source/data'
make[2]: Leaving directory '/build/source/data'
Making all in man
make[2]: Entering directory '/build/source/man'
/nix/store/ckqaamnjzciyl1wgmh99cjlpdjmzvaiz-libxslt-1.1.42-bin/bin/xsltproc -nonet http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl lxsession.xml
Note: meta manual : no titled ancestor of refentry                 lxsession
Note: meta manual : no refentry/refmeta/refmiscinfo@class=manual   lxsession
Note: meta manual : see http://www.docbook.org/tdg5/en/html/refmi  lxsession
Warn: meta manual : using "http://LXDE.org" for "manual"           lxsession
Warn: AUTHOR sect.: no personblurb|contrib for Ying\-Chun Liu      lxsession
Note: AUTHOR sect.: see http://www.docbook.org/tdg5/en/html/contr  lxsession
Note: AUTHOR sect.: see http://www.docbook.org/tdg5/en/html/perso  lxsession
Note: Writing lxsession.1
/nix/store/ckqaamnjzciyl1wgmh99cjlpdjmzvaiz-libxslt-1.1.42-bin/bin/xsltproc -nonet http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl lxsession-logout.xml        
Note: meta manual : no titled ancestor of refentry                 lxsession-logout
Note: meta manual : no refentry/refmeta/refmiscinfo@class=manual   lxsession-logout
Note: meta manual : see http://www.docbook.org/tdg5/en/html/refmi  lxsession-logout
Warn: meta manual : using "http://LXDE.org" for "manual"           lxsession-logout
Warn: AUTHOR sect.: no personblurb|contrib for Ying\-Chun Liu      lxsession-logout
Note: AUTHOR sect.: see http://www.docbook.org/tdg5/en/html/contr  lxsession-logout
Note: AUTHOR sect.: see http://www.docbook.org/tdg5/en/html/perso  lxsession-logout
Note: Writing lxsession-logout.1
make[2]: Leaving directory '/build/source/man'
Making all in po
make[2]: Entering directory '/build/source/po'
grep: warning: stray \ before #
grep: warning: stray \ before #
grep: warning: stray \ before #
  MSGFMT af.gmo
  MSGFMT am.gmo
  MSGFMT ar.gmo
  MSGFMT ast.gmo
  MSGFMT be.gmo
  MSGFMT bg.gmo
  MSGFMT bn.gmo
  MSGFMT bn_IN.gmo
  MSGFMT ca.gmo
  MSGFMT cs.gmo
  MSGFMT da.gmo
  MSGFMT de.gmo
  MSGFMT el.gmo
  MSGFMT en_GB.gmo
  MSGFMT eo.gmo
  MSGFMT es.gmo
  MSGFMT et.gmo
  MSGFMT eu.gmo
  MSGFMT fa.gmo
  MSGFMT fi.gmo
  MSGFMT fo.gmo
  MSGFMT fr.gmo
  MSGFMT frp.gmo
  MSGFMT gl.gmo
  MSGFMT he.gmo
  MSGFMT hr.gmo
  MSGFMT hu.gmo
  MSGFMT id.gmo
  MSGFMT is.gmo
  MSGFMT it.gmo
  MSGFMT ja.gmo
  MSGFMT kk.gmo
  MSGFMT ko.gmo
  MSGFMT lg.gmo
  MSGFMT lt.gmo
  MSGFMT ml.gmo
  MSGFMT ms.gmo
  MSGFMT nb.gmo
  MSGFMT nl.gmo
  MSGFMT nn.gmo
  MSGFMT pa.gmo
  MSGFMT pl.gmo
  MSGFMT ps.gmo
  MSGFMT pt.gmo
  MSGFMT pt_BR.gmo
  MSGFMT ro.gmo
  MSGFMT ru.gmo
  MSGFMT si.gmo
  MSGFMT sk.gmo
  MSGFMT sl.gmo
  MSGFMT sr.gmo
  MSGFMT sr@latin.gmo
  MSGFMT sv.gmo
  MSGFMT te.gmo
  MSGFMT th.gmo
  MSGFMT tr.gmo
  MSGFMT tt_RU.gmo
  MSGFMT ug.gmo
  MSGFMT uk.gmo
  MSGFMT ur.gmo
  MSGFMT ur_PK.gmo
  MSGFMT vi.gmo
  MSGFMT zh_CN.gmo
  MSGFMT zh_TW.gmo
make[2]: Leaving directory '/build/source/po'
make[2]: Entering directory '/build/source'
  VALAC    lxsettings_daemon_lxsettings_daemon_vala.stamp
  CC       lxsettings-daemon/lxsettings_daemon-main.o
  CC       lxsettings-daemon/lxsettings_daemon-settings-daemon.o
  CC       lxsettings-daemon/lxsettings_daemon-xevent.o
  CC       lxsettings-daemon/lxsettings_daemon-xsettings-common.o
  CC       lxsettings-daemon/lxsettings_daemon-xsettings-manager.o
  CC       lxsettings-daemon/lxsettings_daemon-xutils.o
  CCLD     lxsettings-daemon/lxsettings-daemon
  VALAC    xdg_autostart_lxsession_xdg_autostart_vala.stamp
  CC       xdg-autostart/lxsession_xdg_autostart-main.o
  CC       xdg-autostart/lxsession_xdg_autostart-xdg-autostart.o
  CCLD     xdg-autostart/lxsession-xdg-autostart
  VALAC    lxsession_lxsession_vala.stamp
[001mgtk+-2.0.vapi:5873.40-5873.42:[0m [001;35mwarning:[0m [001m`Gtk'[0m is deprecated. Use gtk+-3.0
 5873 | 		public Window (Gtk.WindowType type = Gtk.WindowType.TOPLEVEL);
      | [001;32m		                                     ^~~                      [0m
[001mlxsession/dbus-common.vala:80.5-80.32:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   80 |     public abstract void restart () throws IOError;
      | [001;32m    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~                   [0m
[001mlxsession/dbus-common.vala:81.5-81.29:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   81 |     public abstract void stop () throws IOError;
      | [001;32m    ^~~~~~~~~~~~~~~~~~~~~~~~~                   [0m
[001mlxsession/dbus-common.vala:82.5-82.42:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   82 |     public abstract async bool can_restart () throws IOError;
      | [001;32m    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                   [0m
[001mlxsession/dbus-common.vala:83.5-83.39:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   83 |     public abstract async bool can_stop () throws IOError;
      | [001;32m    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                   [0m
[001mlxsession/main.vala:120.9-120.11:[0m [001;35mwarning:[0m [001m`Gtk'[0m is deprecated. Use gtk+-3.0
  120 |         Gtk.init (ref args);
      | [001;32m        ^~~                 [0m
[001mlxsession/dbus-lxde-session.vala:26.9-26.28:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   26 |         public void Shutdown()
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~  [0m
[001mlxsession/dbus-lxde-session.vala:31.9-31.37:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   31 |         public async void CanShutdown(out bool is_available)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~                       [0m
[001mlxsession/dbus-lxde-session.vala:37.9-37.35:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   37 |         public void RequestShutdown()
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~  [0m
[001mlxsession/dbus-lxde-session.vala:43.9-43.33:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   43 |         public void RequestReboot()
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~  [0m
[001mlxsession/dbus-lxde-session.vala:49.9-49.26:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   49 |         public void Logout()
      | [001;32m        ^~~~~~~~~~~~~~~~~~  [0m
[001mlxsession/dbus-lxde-session.vala:55.9-55.40:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   55 |         public void ReloadSettingsDaemon()
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  [0m
[001mlxsession/dbus-lxde-session.vala:62.9-62.34:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   62 |         public void SessionSupport (out string[] list)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~                    [0m
[001mlxsession/dbus-lxde-session.vala:67.9-67.40:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   67 |         public void SessionSupportDetail (string key1, out string[] list)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                 [0m
[001mlxsession/dbus-lxde-session.vala:72.9-72.30:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   72 |         public void SessionGet(string key1, string key2, out string command)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~                                              [0m
[001mlxsession/dbus-lxde-session.vala:77.9-77.30:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   77 |         public void SessionSet(string key1, string key2, string command_to_set)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~                                                 [0m
[001mlxsession/dbus-lxde-session.vala:83.9-83.36:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   83 |         public void XsettingsSupport (out string[] list)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~                    [0m
[001mlxsession/dbus-lxde-session.vala:88.9-88.42:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   88 |         public void XsettingsSupportDetail (string key1, out string[] list)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                 [0m
[001mlxsession/dbus-lxde-session.vala:93.9-93.32:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   93 |         public void XsettingsGet(string key1, string key2, out string command)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~                                              [0m
[001mlxsession/dbus-lxde-session.vala:98.9-98.32:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   98 |         public void XsettingsSet(string key1, string key2, string command_to_set)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~                                                 [0m
[001mlxsession/dbus-lxde-session.vala:104.9-104.32:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  104 |         public void StateSupport (out string[] list)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~                    [0m
[001mlxsession/dbus-lxde-session.vala:109.9-109.38:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  109 |         public void StateSupportDetail (string key1, out string[] list)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                 [0m
[001mlxsession/dbus-lxde-session.vala:114.9-114.28:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  114 |         public void StateGet(string key1, string key2, out string command)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~                                              [0m
[001mlxsession/dbus-lxde-session.vala:119.9-119.28:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  119 |         public void StateSet(string key1, string key2, string command_to_set)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~                                                 [0m
[001mlxsession/dbus-lxde-session.vala:125.9-125.31:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  125 |         public void DbusSupport (out string[] list)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~                    [0m
[001mlxsession/dbus-lxde-session.vala:130.9-130.37:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  130 |         public void DbusSupportDetail (string key1, out string[] list)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                 [0m
[001mlxsession/dbus-lxde-session.vala:135.9-135.27:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  135 |         public void DbusGet(string key1, string key2, out string command)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~                                              [0m
[001mlxsession/dbus-lxde-session.vala:140.9-140.27:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  140 |         public void DbusSet(string key1, string key2, string command_to_set)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~                                                 [0m
[001mlxsession/dbus-lxde-session.vala:146.9-146.33:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  146 |         public void KeymapSupport (out string[] list)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~                    [0m
[001mlxsession/dbus-lxde-session.vala:151.9-151.39:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  151 |         public void KeymapSupportDetail (string key1, out string[] list)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                 [0m
[001mlxsession/dbus-lxde-session.vala:156.9-156.29:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  156 |         public void KeymapGet(string key1, string key2, out string command)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~                                              [0m
[001mlxsession/dbus-lxde-session.vala:161.9-161.29:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  161 |         public void KeymapSet(string key1, string key2, string command_to_set)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~                                                 [0m
[001mlxsession/dbus-lxde-session.vala:166.9-166.34:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  166 |         public void KeymapActivate()
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~  [0m
[001mlxsession/dbus-lxde-session.vala:188.9-188.38:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  188 |         public void EnvironmentSupport (out string[] list)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                    [0m
[001mlxsession/dbus-lxde-session.vala:193.9-193.44:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  193 |         public void EnvironmentSupportDetail (string key1, out string[] list)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                 [0m
[001mlxsession/dbus-lxde-session.vala:198.9-198.34:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  198 |         public void EnvironmentGet(string key1, string key2, out string command)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~                                              [0m
[001mlxsession/dbus-lxde-session.vala:203.9-203.34:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  203 |         public void EnvironmentSet(string key1, string key2, string command_to_set)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~                                                 [0m
[001mlxsession/dbus-lxde-session.vala:673.9-673.32:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  673 |         public void A11yActivate()
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~  [0m
[001mlxsession/dbus-lxde-session.vala:652.9-652.33:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  652 |         public void ProxyActivate()
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~  [0m
[001mlxsession/dbus-lxde-session.vala:631.9-631.34:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  631 |         public void XrandrActivate()
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~  [0m
[001mlxsession/dbus-lxde-session.vala:610.9-610.35:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  610 |         public void KeyringActivate()
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~  [0m
[001mlxsession/dbus-lxde-session.vala:568.9-568.40:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  568 |         public void CrashManagerActivate()
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  [0m
[001mlxsession/dbus-lxde-session.vala:589.9-589.42:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  589 |         public void CrashManagerInactivate()
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  [0m
[001mlxsession/dbus-lxde-session.vala:303.9-303.33:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  303 |         public void SessionLaunch(string name, string option)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~                            [0m
[001mlxsession/dbus-lxde-session.vala:547.9-547.38:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  547 |         public void CrashManagerLaunch()
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  [0m
[001mlxsession/dbus-lxde-session.vala:1131.9-1131.47:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
 1131 |         public async void PackageManagerRunning (out bool is_running)
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                      [0m
[001mlxsession/dbus-lxde-session.vala:1138.9-1138.40:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
 1138 |         public void TestIconNotification()
      | [001;32m        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  [0m
[001mlxsession/dbus-gnome-session.vala:47.5-47.22:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   47 |     public void Setenv( string value ) {
      | [001;32m    ^~~~~~~~~~~~~~~~~~                  [0m
[001mlxsession/dbus-gnome-session.vala:54.5-54.35:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   54 |     public void InitializationError ( string mess, bool fatal ) {
      | [001;32m    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                              [0m
[001mlxsession/dbus-gnome-session.vala:65.5-65.36:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   65 |     public async void RegisterClient(string app_id, string client_startup_id) {
      | [001;32m    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                           [0m
[001mlxsession/dbus-gnome-session.vala:79.5-79.38:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   79 |     public async void UnregisterClient() {
      | [001;32m    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~    [0m
[001mlxsession/dbus-gnome-session.vala:89.5-89.29:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
   89 |     public async void Inhibit(string app_id, uint toplevel_xid, string reason, uint flags, out uint inhibit_cookie)
      | [001;32m    ^~~~~~~~~~~~~~~~~~~~~~~~~                                                                                      [0m
[001mlxsession/dbus-gnome-session.vala:122.5-122.31:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  122 |     public async void Uninhibit(uint inhibit_cookie)
      | [001;32m    ^~~~~~~~~~~~~~~~~~~~~~~~~~~                     [0m
[001mlxsession/dbus-gnome-session.vala:207.5-207.24:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  207 |     public void Shutdown() {
      | [001;32m    ^~~~~~~~~~~~~~~~~~~~    [0m
[001mlxsession/dbus-gnome-session.vala:211.5-211.33:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  211 |     public async void CanShutdown(out bool is_available) {
      | [001;32m    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~                         [0m
[001mlxsession/dbus-gnome-session.vala:216.5-216.22:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  216 |     public void Logout(uint mode) {
      | [001;32m    ^~~~~~~~~~~~~~~~~~             [0m
[001mlxsession/dbus-gnome-session.vala:229.5-229.32:[0m [001;35mwarning:[0m DBus methods are recommended to throw at least [001m`GLib.Error'[0m or [001m`GLib.DBusError, GLib.IOError'[0m
  229 |     public void IsSessionRunning(out bool running )
      | [001;32m    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~                   [0m
[001mlxsession/app.vala:1895.20-1895.26:[0m [001;35mwarning:[0m Local variable [001m`command'[0m declared but never used
 1895 |             string command =  "/usr/bin/pkexec /usr/share/apport/apport-gtk" + " " + this.crash_dir_path + crash_list.nth_data(last_item) ;
      | [001;32m                   ^~~~~~~                                                                                                                 [0m
[001mlxsession/app.vala:1898.20-1898.32:[0m [001;35mwarning:[0m Local variable [001m`launch_string'[0m declared but never used
 1898 |             string launch_string = "Report last crash";
      | [001;32m                   ^~~~~~~~~~~~~                       [0m
Compilation succeeded - 59 warning(s)
  CC       lxsession/lxsession-main.o
  CC       lxsession/lxsession-app.o
  CC       lxsession/lxsession-autostart.o
  CC       lxsession/lxsession-conffiles.o
  CC       lxsession/lxsession-control.o
  CC       lxsession/lxsession-dbus-common.o
  CC       lxsession/lxsession-dbus-gnome-session.o
  CC       lxsession/lxsession-dbus-lxde-session.o
  CC       lxsession/lxsession-environement.o
  CC       lxsession/lxsession-notifications.o
  CC       lxsession/lxsession-options.o
  CC       lxsession/lxsession-process.o
  CC       lxsession/lxsession-settings.o
  CC       lxsession/lxsession-utils.o
  CC       lxsettings-daemon/lxsession_lxsession-settings-daemon.o
  CC       lxsettings-daemon/lxsession_lxsession-xevent.o
  CC       lxsettings-daemon/lxsession_lxsession-xsettings-common.o
  CC       lxsettings-daemon/lxsession_lxsession-xsettings-manager.o
  CC       lxsettings-daemon/lxsession_lxsession-xutils.o
  CCLD     lxsession/lxsession
  VALAC    lxpolkit_lxpolkit_vala.stamp
[001mgtk+-2.0.vapi:5873.40-5873.42:[0m [001;35mwarning:[0m [001m`Gtk'[0m is deprecated. Use gtk+-3.0
 5873 | 		public Window (Gtk.WindowType type = Gtk.WindowType.TOPLEVEL);
      | [001;32m		                                     ^~~                      [0m
[001mlxpolkit/main.vala:32.13-32.15:[0m [001;35mwarning:[0m [001m`Gtk'[0m is deprecated. Use gtk+-3.0
   32 |             Gtk.init (ref args);
      | [001;32m            ^~~                 [0m
[001mlxpolkit/main.vala:36.13-36.27:[0m [001;35mwarning:[0m unhandled error [001m`GLib.Error'[0m
   36 |             app.register ();
      | [001;32m            ^~~~~~~~~~~~~~~ [0m
Compilation succeeded - 3 warning(s)
  CC       lxpolkit/lxpolkit-main.o
  CC       lxpolkit/lxpolkit-lxpolkit.o
  CC       lxpolkit/lxpolkit-lxpolkit-listener.o
[01m[Klxpolkit/lxpolkit-listener.c:[m[K In function '[01m[Kon_completed[m[K':
[01m[Klxpolkit/lxpolkit-listener.c:88:22:[m[K [01;31m[Kerror: [m[Kpassing argument 1 of '[01m[Kshow_msg[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
   88 |         show_msg([01;31m[Kdata->dlg[m[K, GTK_MESSAGE_ERROR, _("Authentication failed!\nWrong password?"));
      |                  [01;31m[K~~~~^~~~~[m[K
      |                      [01;31m[K|[m[K
      |                      [01;31m[KGtkWidget * {aka struct _GtkWidget *}[m[K
In file included from [01m[Klxpolkit/lxpolkit-listener.c:27[m[K:
[01m[Klxpolkit/lxpolkit.h:30:26:[m[K [01;36m[Knote: [m[Kexpected '[01m[KGtkWindow *[m[K' {aka '[01m[Kstruct _GtkWindow *[m[K'} but argument is of type '[01m[KGtkWidget *[m[K' {aka '[01m[Kstruct _GtkWidget *[m[K'}
   30 | void show_msg([01;36m[KGtkWindow* parent[m[K, GtkMessageType type, const char* msg);
      |               [01;36m[K~~~~~~~~~~~^~~~~~[m[K
[01m[Klxpolkit/lxpolkit-listener.c:92:32:[m[K [01;31m[Kerror: [m[Kpassing argument 1 of '[01m[Kgtk_entry_set_text[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
   92 |         gtk_entry_set_text([01;31m[Kdata->request[m[K, "");
      |                            [01;31m[K~~~~^~~~~~~~~[m[K
      |                                [01;31m[K|[m[K
      |                                [01;31m[KGtkWidget * {aka struct _GtkWidget *}[m[K
In file included from [01m[K/nix/store/mgk8ipxsq47kq754lp0dnfhhxfq7gkpj-gtk+-2.24.33-dev/include/gtk-2.0/gtk/gtktreeview.h:31[m[K,
                 from [01m[K/nix/store/mgk8ipxsq47kq754lp0dnfhhxfq7gkpj-gtk+-2.24.33-dev/include/gtk-2.0/gtk/gtkcombobox.h:29[m[K,
                 from [01m[K/nix/store/mgk8ipxsq47kq754lp0dnfhhxfq7gkpj-gtk+-2.24.33-dev/include/gtk-2.0/gtk/gtk.h:72[m[K,
                 from [01m[Klxpolkit/lxpolkit.h:26[m[K:
[01m[K/nix/store/mgk8ipxsq47kq754lp0dnfhhxfq7gkpj-gtk+-2.24.33-dev/include/gtk-2.0/gtk/gtkentry.h:213:65:[m[K [01;36m[Knote: [m[Kexpected '[01m[KGtkEntry *[m[K' {aka '[01m[Kstruct _GtkEntry *[m[K'} but argument is of type '[01m[KGtkWidget *[m[K' {aka '[01m[Kstruct _GtkWidget *[m[K'}
  213 | void       gtk_entry_set_text                   ([01;36m[KGtkEntry      *entry[m[K,
      |                                                  [01;36m[K~~~~~~~~~~~~~~~^~~~~[m[K
[01m[Klxpolkit/lxpolkit-listener.c:94:29:[m[K [01;31m[Kerror: [m[Kpassing argument 1 of '[01m[Kon_user_changed[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
   94 |         on_user_changed([01;31m[Kdata->id[m[K, data);
      |                         [01;31m[K~~~~^~~~[m[K
      |                             [01;31m[K|[m[K
      |                             [01;31m[KGtkWidget * {aka struct _GtkWidget *}[m[K
[01m[Klxpolkit/lxpolkit-listener.c:65:42:[m[K [01;36m[Knote: [m[Kexpected '[01m[KGtkComboBox *[m[K' {aka '[01m[Kstruct _GtkComboBox *[m[K'} but argument is of type '[01m[KGtkWidget *[m[K' {aka '[01m[Kstruct _GtkWidget *[m[K'}
   65 | static void on_user_changed([01;36m[KGtkComboBox* id_combo[m[K, DlgData* data);
      |                             [01;36m[K~~~~~~~~~~~~~^~~~~~~~[m[K
[01m[Klxpolkit/lxpolkit-listener.c:[m[K In function '[01m[Kon_request[m[K':
[01m[Klxpolkit/lxpolkit-listener.c:109:28:[m[K [01;31m[Kerror: [m[Kpassing argument 1 of '[01m[Kgtk_label_set_text[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
  109 |     gtk_label_set_text([01;31m[Kdata->request_label[m[K, msg);
      |                        [01;31m[K~~~~^~~~~~~~~~~~~~~[m[K
      |                            [01;31m[K|[m[K
      |                            [01;31m[KGtkWidget * {aka struct _GtkWidget *}[m[K
In file included from [01m[K/nix/store/mgk8ipxsq47kq754lp0dnfhhxfq7gkpj-gtk+-2.24.33-dev/include/gtk-2.0/gtk/gtkaccellabel.h:38[m[K,
                 from [01m[K/nix/store/mgk8ipxsq47kq754lp0dnfhhxfq7gkpj-gtk+-2.24.33-dev/include/gtk-2.0/gtk/gtk.h:35[m[K:
[01m[K/nix/store/mgk8ipxsq47kq754lp0dnfhhxfq7gkpj-gtk+-2.24.33-dev/include/gtk-2.0/gtk/gtklabel.h:112:67:[m[K [01;36m[Knote: [m[Kexpected '[01m[KGtkLabel *[m[K' {aka '[01m[Kstruct _GtkLabel *[m[K'} but argument is of type '[01m[KGtkWidget *[m[K' {aka '[01m[Kstruct _GtkWidget *[m[K'}
  112 | void                  gtk_label_set_text          ([01;36m[KGtkLabel      *label[m[K,
      |                                                    [01;36m[K~~~~~~~~~~~~~~~^~~~~[m[K
[01m[Klxpolkit/lxpolkit-listener.c:110:34:[m[K [01;31m[Kerror: [m[Kpassing argument 1 of '[01m[Kgtk_entry_set_visibility[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
  110 |     gtk_entry_set_visibility([01;31m[Kdata->request[m[K, echo_on);
      |                              [01;31m[K~~~~^~~~~~~~~[m[K
      |                                  [01;31m[K|[m[K
      |                                  [01;31m[KGtkWidget * {aka struct _GtkWidget *}[m[K
[01m[K/nix/store/mgk8ipxsq47kq754lp0dnfhhxfq7gkpj-gtk+-2.24.33-dev/include/gtk-2.0/gtk/gtkentry.h:176:65:[m[K [01;36m[Knote: [m[Kexpected '[01m[KGtkEntry *[m[K' {aka '[01m[Kstruct _GtkEntry *[m[K'} but argument is of type '[01m[KGtkWidget *[m[K' {aka '[01m[Kstruct _GtkWidget *[m[K'}
  176 | void       gtk_entry_set_visibility             ([01;36m[KGtkEntry      *entry[m[K,
      |                                                  [01;36m[K~~~~~~~~~~~~~~~^~~~~[m[K
[01m[Klxpolkit/lxpolkit-listener.c:[m[K In function '[01m[Kon_show_error[m[K':
[01m[Klxpolkit/lxpolkit-listener.c:116:18:[m[K [01;31m[Kerror: [m[Kpassing argument 1 of '[01m[Kshow_msg[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
  116 |     show_msg([01;31m[Kdata->dlg[m[K, GTK_MESSAGE_ERROR, text);
      |              [01;31m[K~~~~^~~~~[m[K
      |                  [01;31m[K|[m[K
      |                  [01;31m[KGtkWidget * {aka struct _GtkWidget *}[m[K
[01m[Klxpolkit/lxpolkit.h:30:26:[m[K [01;36m[Knote: [m[Kexpected '[01m[KGtkWindow *[m[K' {aka '[01m[Kstruct _GtkWindow *[m[K'} but argument is of type '[01m[KGtkWidget *[m[K' {aka '[01m[Kstruct _GtkWidget *[m[K'}
   30 | void show_msg([01;36m[KGtkWindow* parent[m[K, GtkMessageType type, const char* msg);
      |               [01;36m[K~~~~~~~~~~~^~~~~~[m[K
[01m[Klxpolkit/lxpolkit-listener.c:[m[K In function '[01m[Kon_show_info[m[K':
[01m[Klxpolkit/lxpolkit-listener.c:122:18:[m[K [01;31m[Kerror: [m[Kpassing argument 1 of '[01m[Kshow_msg[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
  122 |     show_msg([01;31m[Kdata->dlg[m[K, GTK_MESSAGE_INFO, text);
      |              [01;31m[K~~~~^~~~~[m[K
      |                  [01;31m[K|[m[K
      |                  [01;31m[KGtkWidget * {aka struct _GtkWidget *}[m[K
[01m[Klxpolkit/lxpolkit.h:30:26:[m[K [01;36m[Knote: [m[Kexpected '[01m[KGtkWindow *[m[K' {aka '[01m[Kstruct _GtkWindow *[m[K'} but argument is of type '[01m[KGtkWidget *[m[K' {aka '[01m[Kstruct _GtkWidget *[m[K'}
   30 | void show_msg([01;36m[KGtkWindow* parent[m[K, GtkMessageType type, const char* msg);
      |               [01;36m[K~~~~~~~~~~~^~~~~~[m[K
[01m[Klxpolkit/lxpolkit-listener.c:[m[K In function '[01m[Kon_dlg_response[m[K':
[01m[Klxpolkit/lxpolkit-listener.c:130:54:[m[K [01;31m[Kerror: [m[Kpassing argument 1 of '[01m[Kgtk_entry_get_text[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
  130 |         const char* request = gtk_entry_get_text([01;31m[Kdata->request[m[K);
      |                                                  [01;31m[K~~~~^~~~~~~~~[m[K
      |                                                      [01;31m[K|[m[K
      |                                                      [01;31m[KGtkWidget * {aka struct _GtkWidget *}[m[K
[01m[K/nix/store/mgk8ipxsq47kq754lp0dnfhhxfq7gkpj-gtk+-2.24.33-dev/include/gtk-2.0/gtk/gtkentry.h:216:65:[m[K [01;36m[Knote: [m[Kexpected '[01m[KGtkEntry *[m[K' {aka '[01m[Kstruct _GtkEntry *[m[K'} but argument is of type '[01m[KGtkWidget *[m[K' {aka '[01m[Kstruct _GtkWidget *[m[K'}
  216 | const gchar* gtk_entry_get_text                 ([01;36m[KGtkEntry      *entry[m[K);
      |                                                  [01;36m[K~~~~~~~~~~~~~~~^~~~~[m[K
[01m[Klxpolkit/lxpolkit-listener.c:[m[K In function '[01m[Kinitiate_authentication[m[K':
[01m[Klxpolkit/lxpolkit-listener.c:198:46:[m[K [01;31m[Kerror: [m[Kpassing argument 1 of '[01m[Kg_simple_async_result_new[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
  198 |     data->result = g_simple_async_result_new([01;31m[Klistener[m[K, callback, user_data, initiate_authentication);
      |                                              [01;31m[K^~~~~~~~[m[K
      |                                              [01;31m[K|[m[K
      |                                              [01;31m[KPolkitAgentListener * {aka struct _PolkitAgentListener *}[m[K
In file included from [01m[K/nix/store/y8fxzs8srzd6d74a85zbpssjr8wkj9q4-glib-2.82.1-dev/include/glib-2.0/gio/gio.h:141[m[K,
                 from [01m[K/nix/store/d4izmmkdv66m7hd8r8y0kdniz7p9jdc4-polkit-124-dev/include/polkit-1/polkit/polkitactiondescription.h:30[m[K,
                 from [01m[K/nix/store/d4izmmkdv66m7hd8r8y0kdniz7p9jdc4-polkit-124-dev/include/polkit-1/polkit/polkit.h:30[m[K,
                 from [01m[K/nix/store/d4izmmkdv66m7hd8r8y0kdniz7p9jdc4-polkit-124-dev/include/polkit-1/polkitagent/polkitagentlistener.h:29[m[K,
                 from [01m[K/nix/store/d4izmmkdv66m7hd8r8y0kdniz7p9jdc4-polkit-124-dev/include/polkit-1/polkitagent/polkitagent.h:32[m[K,
                 from [01m[Klxpolkit/lxpolkit-listener.h:27[m[K,
                 from [01m[Klxpolkit/lxpolkit-listener.c:26[m[K:
[01m[K/nix/store/y8fxzs8srzd6d74a85zbpssjr8wkj9q4-glib-2.82.1-dev/include/glib-2.0/gio/gsimpleasyncresult.h:48:86:[m[K [01;36m[Knote: [m[Kexpected '[01m[KGObject *[m[K' {aka '[01m[Kstruct _GObject *[m[K'} but argument is of type '[01m[KPolkitAgentListener *[m[K' {aka '[01m[Kstruct _PolkitAgentListener *[m[K'}
   48 | GSimpleAsyncResult *g_simple_async_result_new              ([01;36m[KGObject                 *source_object[m[K,
      |                                                             [01;36m[K~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~[m[K
[01m[Klxpolkit/lxpolkit-listener.c:263:37:[m[K [01;31m[Kerror: [m[Kpassing argument 1 of '[01m[Kgtk_combo_box_set_model[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
  263 |         gtk_combo_box_set_model([01;31m[Kdata->id[m[K, GTK_TREE_MODEL(store));
      |                                 [01;31m[K~~~~^~~~[m[K
      |                                     [01;31m[K|[m[K
      |                                     [01;31m[KGtkWidget * {aka struct _GtkWidget *}[m[K
[01m[K/nix/store/mgk8ipxsq47kq754lp0dnfhhxfq7gkpj-gtk+-2.24.33-dev/include/gtk-2.0/gtk/gtkcombobox.h:109:64:[m[K [01;36m[Knote: [m[Kexpected '[01m[KGtkComboBox *[m[K' {aka '[01m[Kstruct _GtkComboBox *[m[K'} but argument is of type '[01m[KGtkWidget *[m[K' {aka '[01m[Kstruct _GtkWidget *[m[K'}
  109 | void          gtk_combo_box_set_model        ([01;36m[KGtkComboBox     *combo_box[m[K,
      |                                               [01;36m[K~~~~~~~~~~~~~~~~~^~~~~~~~~[m[K
[01m[Klxpolkit/lxpolkit-listener.c:266:38:[m[K [01;31m[Kerror: [m[Kpassing argument 1 of '[01m[Kgtk_combo_box_set_active[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
  266 |         gtk_combo_box_set_active([01;31m[Kdata->id[m[K, 0);
      |                                  [01;31m[K~~~~^~~~[m[K
      |                                      [01;31m[K|[m[K
      |                                      [01;31m[KGtkWidget * {aka struct _GtkWidget *}[m[K
[01m[K/nix/store/mgk8ipxsq47kq754lp0dnfhhxfq7gkpj-gtk+-2.24.33-dev/include/gtk-2.0/gtk/gtkcombobox.h:101:64:[m[K [01;36m[Knote: [m[Kexpected '[01m[KGtkComboBox *[m[K' {aka '[01m[Kstruct _GtkComboBox *[m[K'} but argument is of type '[01m[KGtkWidget *[m[K' {aka '[01m[Kstruct _GtkWidget *[m[K'}
  101 | void          gtk_combo_box_set_active       ([01;36m[KGtkComboBox     *combo_box[m[K,
      |                                               [01;36m[K~~~~~~~~~~~~~~~~~^~~~~~~~~[m[K
make[2]: *** [Makefile:1573: lxpolkit/lxpolkit-lxpolkit-listener.o] Error 1
make[2]: Leaving directory '/build/source'
make[1]: *** [Makefile:2380: all-recursive] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:1051: all] Error 2
```

</details>

## Additional context

This is caused by the GCC 14 upgrade.

A [bug report](https://sourceforge.net/p/lxde/bugs/973/) has been filed upstream.

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/desktops/lxde/core/lxsession/default.nix
+++ b/pkgs/desktops/lxde/core/lxsession/default.nix
@@ -1,6 +1,7 @@
 {
   lib,
   stdenv,
+  fetchpatch,
   fetchFromGitHub,
   autoconf,
   automake,
@@ -28,7 +29,17 @@ stdenv.mkDerivation rec {
     sha256 = "17sqsx57ymrimm5jfmcyrp7b0nzi41bcvpxsqckmwbhl19g6c17d";
   };
 
-  patches = [ ./xmlcatalog_patch.patch ];
+  patches = [
+    ./xmlcatalog_patch.patch
+
+    # lxsession compilation is broken upstream as of GCC 14
+    # https://sourceforge.net/p/lxde/bugs/973/
+    (fetchpatch {
+      name = "0001-Fix-build-on-GCC-14.patch";
+      url = "https://sourceforge.net/p/lxde/bugs/973/attachment/0001-Fix-build-on-GCC-14.patch";
+      hash = "sha256-lxF3HZy5uLK7Cfu8W1A03syZf7OWXpHiU2Fk+xBl39g=";
+    })
+  ];
 
   nativeBuildInputs = [
     autoconf

